### PR TITLE
SortPetCommand fixes and tests

### PR DESCRIPTION
**Pull request by md-azsa**
Fixes issue #154

### CODE REVIEW

File: src/main/java/seedu/address/model/Model.java [hunk 1]
```diff
@@ -147,7 +148,7 @@ void updatePerson(Person target, Person editedPerson)
     /**
      * Sorts the pet list.
      */
-    void sortPetList();
+    void sortPetList() throws ClientPetAssociationListEmptyException;
```
Comment (jonathanwj) [inline src/main/java/seedu/address/model/Model.java]: Maybe header comments can be more specific about how its sorted.

File: src/main/java/seedu/address/model/association/exceptions/ClientPetAssociationListEmptyException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package seedu.address.model.association.exceptions;
+
+/**
+ * Exception when the client list is empty and user tries to sort
```
Comment (jonathanwj) [inline src/main/java/seedu/address/model/association/exceptions/ClientPetAssociationListEmptyException.java]: the header comment need not so specific to sort.
I think this exception can be used for other functions other than sort.

File: src/main/java/seedu/address/model/Model.java (hunk 1 above)
Comment (md-azsa) [reply src/main/java/seedu/address/model/Model.java]: alright will issue the amendments by tonight! 👍 thanks!

File: src/main/java/seedu/address/model/association/exceptions/ClientPetAssociationListEmptyException.java (hunk 2 above)
Comment (md-azsa) [reply src/main/java/seedu/address/model/association/exceptions/ClientPetAssociationListEmptyException.java]: alright will issue the amendments by tonight! 👍 thanks! 😄

File: src/main/java/seedu/address/model/Model.java (hunk 1 above)
Comment (md-azsa) [reply src/main/java/seedu/address/model/Model.java]: resolved

**Pull request merged**